src/Razor/src/Microsoft.VisualStudio.LiveShare.Razor/Guest/RazorGuestInitializationService.cs

```diff
@@ -52,14 +60,28 @@ private async Task EnsureViewImportsCopiedAsync(CollaborationSession sessionCont
             };
 
             var copyTasks = new List<Task>();
-            var roots = await sessionContext.ListRootsAsync(cancellationToken);
-            foreach (var root in roots)
+
+            try
             {
-                var fileUris = await sessionContext.ListDirectoryAsync(root, listDirectoryOptions, cancellationToken);
-                StartViewImportsCopy(fileUris, copyTasks, sessionContext, cancellationToken);
-            }
+                var roots = await sessionContext.ListRootsAsync(cancellationToken);
+
+                if (cancellationToken.IsCancellationRequested)
+                {
+                    return;
+                }
 
-            await Task.WhenAll(copyTasks);
+                foreach (var root in roots)
+                {
+                    var fileUris = await sessionContext.ListDirectoryAsync(root, listDirectoryOptions, cancellationToken);
+                    StartViewImportsCopy(fileUris, copyTasks, sessionContext, cancellationToken);
```

Comment:
Super unrelated, does the path comparison on Line 91 need a `StringComparison.OrdinalIgnoreCase` check?

Reply:
No because we enforce the casing of _ViewImports